src/uct/api/v2/uct_v2.h

```diff
@@ -274,23 +274,43 @@ typedef enum {
     /**
      * The flag is used indicate that remote access to a memory region
      * associated with the remote key must fail once the memory region is
-     * deregister using @ref uct_md_mem_dereg_v2 with
+     * deregistered using @ref uct_md_mem_dereg_v2 with
      * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set. Using
      * @ref uct_md_mem_dereg_v2 deregistration routine with
      * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set on an rkey that was
-     * generated without @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag will
-     * not function correctly, and may result in data corruption. In other words
-     * in order for @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag to function
-     * the @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag must be set.
+     * generated without
+     * @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE_RMA and/or
+     * @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE_AMO flags will not function
+     * correctly, and may result in data corruption. In other words in order for
+     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag to function
+     * the @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE_RMA and/or
+     * the @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE_AMO flag must be set.
      */
-    UCT_MD_MKEY_PACK_FLAG_INVALIDATE = UCS_BIT(0),
+    UCT_MD_MKEY_PACK_FLAG_INVALIDATE_RMA = UCS_BIT(0),
+
+    /**
+     * The flag is used indicate that remote atomic operation to a memory region
+     * associated with the remote key must fail once the memory region is
+     * deregistered using @ref uct_md_mem_dereg_v2 with
+     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set. Using
```

Comment:
This does not read right. I suggest the following (@gitrepoidoscar  please review):

The flag is used indicate that access to a memory region using remote atomic operations must fail once the memory region is deregistered using @ref uct_md_mem_dereg_v2 with @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set,

Reply:
fixed

Reply:
@shamisp could you please have another look?